session.go

```diff
@@ -275,14 +267,13 @@ func (s *Session) exitErr(err error) {
 // GoAway can be used to prevent accepting further
 // connections. It does not close the underlying conn.
 func (s *Session) GoAway() error {
-	return s.waitForSend(s.goAway(goAwayNormal), nil)
+	return s.sendMsg(s.goAway(goAwayNormal), nil, nil)
```

Comment:
There is now only one way to send things (all sends are async, like the network, dammit!).